crates/uv-fs/src/locked_file.rs

```diff
@@ -0,0 +1,348 @@
+use std::fmt::Display;
+use std::path::{Path, PathBuf};
+use std::sync::LazyLock;
+use std::time::Duration;
+use std::{env, io, thread};
+
+use fs2::FileExt;
+use thiserror::Error;
+use tracing::{debug, error, info, trace, warn};
+
+use uv_static::EnvVars;
+
+use crate::Simplified;
+
+/// Parsed value of `UV_LOCK_TIMEOUT`, with a default of 5 min.
+static LOCK_TIMEOUT: LazyLock<Duration> = LazyLock::new(|| {
+    let default_timeout = Duration::from_secs(300);
+    let Some(lock_timeout) = env::var_os(EnvVars::UV_LOCK_TIMEOUT) else {
+        return default_timeout;
+    };
+
+    if let Some(lock_timeout) = lock_timeout
+        .to_str()
+        .and_then(|lock_timeout| lock_timeout.parse::<u64>().ok())
+    {
+        Duration::from_secs(lock_timeout)
+    } else {
+        warn!(
+            "Could not parse value of {} as integer: {:?}",
+            EnvVars::UV_LOCK_TIMEOUT,
+            lock_timeout
+        );
+        default_timeout
+    }
+});
+
+#[derive(Debug, Error)]
+pub enum LockedFileError {
+    #[error(
+        "Timeout ({}s) when waiting for lock on `{}` at `{}`, is another uv process running? Set `{}` to increase the timeout.",
```

Comment:
We might want to say "You can set ... to increase the timeout" instead of "Set" which makes it sounds like you _should_ do that as the solution.